controllers/stories.js

```diff
@@ -0,0 +1,90 @@
+const stories = require("../models/stories");
+/**
+ * Creates new story
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - Story object
+ * @param res {Object} - Express response object
+ */
+const addNewStory = async (req, res) => {
+  try {
+    const story = await stories.addOrUpdateStory(req.body);
+
+    if (story) {
+      return res.json({
+        message: "Story created successfully!",
+        story: story.storyDetails,
+        id: story.storyId,
+      });
+    } else {
+      logger.error("Error while creating new story: Incorrect username passed");
+      return res.boom.badRequest("Unable to add story");
+    }
+  } catch (err) {
+    logger.error(`Error while creating new story: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+/**
+ * Fetches all the stories
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const fetchStories = async (req, res) => {
+  try {
+    const result = {};
+    if (req.params.id) {
+      const story = await stories.fetchStory(req.params.id);
+      if (!story.storyData) {
+        return res.boom.notFound("Story not found");
+      }
+      result.message = "Story returned successfully!";
+      result.story = story.storyData;
+    } else {
+      const allStories = await stories.fetchStories(req.query);
+      if (!allStories) {
+        logger.error("Error while fetching story: Incorrect query parameters");
+        return res.boom.badRequest("Unable to fetch stories");
+      }
+      result.message = "Stories returned successfully!";
+      result.stories = allStories.length > 0 ? allStories : [];
+    }
+    return res.json({ ...result });
+  } catch (err) {
+    logger.error(`Error while fetching stories ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
```

Comment:
Can we have two different controller function? `fetchStory` and `fetchStories` that links to `GET story/:id` and `GET story/` respectively.

Reply:
Thanks for the suggestion. Made the required changes